### ISSUE TITLE
Parameter override

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#687]
This inspection should warn if parameter is overrided in some point of method or function.

**Example:**

```php
function ($name = null) {
    $name = $name ?? 'John';
    // Warn: $name parameter was overrided
}
```

**Solution:**

```php
function ($inputName = null) {
    $name = $inputName ?? 'John';
}
```

---

The unique exception for this rule is if parameter is a reference.

```php
function (&$name = null) {
    $name = $name ?? 'John';
}
```

---

It is useful to keep the original parameter value in all function structure, so is pratical request it when it is need. This inspection exists for JS, for instance, and I think that is available to Java too.

**Comment by kalessil** [comment]
I need to re-check this - suspicious assignments inspection should report similar cases (but `$name = $name ?? 'John';` is ok as it depends on the argument itself)

**Comment by kalessil** [comment]
Gets reported (variable needs to be used at least twice to get reported, hence the posted sample is not triggering issue)
```php
function a ($param1, $param2) {
    $param1 = 'John';
    $param2 = 'John';
    return [$param1, $param2 = 'John'];
}
```
